src/frontend/js/components/CourseProductItem/PurchasedProductMenu.tsx

```diff
@@ -0,0 +1,135 @@
+import { useSelect } from 'downshift';
+import { useMemo, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Spinner } from 'components/Spinner';
+import { useJoanieApi } from 'data/JoanieApiProvider';
+import type * as Joanie from 'types/Joanie';
+import { handle } from 'utils/errors/handle';
+
+enum MenuItemKey {
+  DOWNLOAD_INVOICE = 'downloadInvoice',
+}
+
+const messages = defineMessages({
+  actionsRelated: {
+    defaultMessage: 'Other actions related to this product',
+    description: 'Accessible label for the purchased product menu',
+    id: 'components.PurchasedProductMenu.actionsRelated',
+  },
+  generatingInvoice: {
+    defaultMessage: 'Invoice is being generated...',
+    description: 'Accessible label when invoice is being generated',
+    id: 'components.PurchasedProductMenu.generatingInvoice',
+  },
+  [MenuItemKey.DOWNLOAD_INVOICE]: {
+    defaultMessage: 'Download invoice',
+    description: 'Label for selector item to download invoice',
+    id: 'components.PurchasedProductMenu.downloadInvoice',
+  },
+});
+
+interface Props {
+  order: Joanie.OrderLite;
+}
+
+const PurchasedProductMenu = ({ order }: Props) => {
+  const API = useJoanieApi();
+  const [loading, setLoading] = useState(false);
+
+  const downloadInvoice = async () => {
+    try {
+      setLoading(true);
+      const $link = document.createElement('a');
+      const file = await API.user.orders.invoice.download({
+        order_id: order.id,
+        invoice_reference: order.main_invoice,
+      });
+      // eslint-disable-next-line compat/compat
+      const url = URL.createObjectURL(file);
+      $link.href = url;
+      $link.download = '';
+
+      const revokeObject = () => {
+        // eslint-disable-next-line compat/compat
+        URL.revokeObjectURL(url);
+        window.removeEventListener('blur', revokeObject);
+      };
+
+      window.addEventListener('blur', revokeObject);
+      $link.click();
+    } catch (error) {
+      handle(error);
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
Because, we generate the document on demand from Joanie, so the response does not contains a link which serve the document but the document itself so we use the trick to create an object URL from frontend to trigger the download.